[18.0][MIG] fs_import_image_advanced*: Migration to 18.0

xaviedoanhduy opened the pull request:
Base on: 

- https://github.com/xaviedoanhduy/storage/pull/1
- `https://github.com/OCA/storage/pull/360`
- `https://github.com/OCA/storage/pull/446`
- `https://github.com/OCA/storage/pull/447`
- `https://github.com/OCA/storage/pull/448`
- `https://github.com/OCA/storage/pull/449`

Note: the red CI is not caused by the 2 modules in this PR. It come from `https://github.com/OCA/storage/pull/449`

ivantodorovich committed:
[IMP] storage_import_image_advanced: action buttons should be in header

ivantodorovich committed:
[FIX] storage_import_image_advanced: Missing filename field

ivantodorovich committed:
[IMP] storage_image_import_advanced: Allow to configure the csv column mapping

HviorForgeFlow committed:
[13.0][ADD] storage_import_image_advanced: Drop shopinvader dependencies + rename from shopinvader_import_image

HviorForgeFlow committed:
Adapt tests

HviorForgeFlow committed:
[IMP] storage_import_image_advanced: black, isort, prettier

JordiBForgeFlow committed:
[storage_import_image_advanced][imp] make the choice of the product identifier more flexible

HviorForgeFlow committed:
[IMP] storage_import_image_advanced: black, isort, prettier

HviorForgeFlow committed:
[FIX] Recompute thumbs explicitly

HviorForgeFlow committed:
[FIX] Zip Filename display

OCA-git-bot committed:
storage_import_image_advanced 13.0.1.0.1

HviorForgeFlow committed:
[FIX] tmpl report_html reference

OCA-git-bot committed:
storage_import_image_advanced 13.0.1.0.2

i-vyshnevska committed:
[MIG][14.0] storage_import_image_advanced

i-vyshnevska committed:
[IMP] storage_import_image_advanced

OCA-git-bot committed:
storage_import_image_advanced 14.0.1.0.1

simahawk committed:
storage_import_image_advanced: fix report render

OCA-git-bot committed:
storage_import_image_advanced 14.0.1.0.2

JasminSForgeFlow committed:
[IMP] storage_import_image_advanced: black, isort, prettier

JasminSForgeFlow committed:
[MIG] storage_import_image_advanced: Migration to 15.0

HviorForgeFlow committed:
[IMP] storage_import_image_advanced: Add hook

lmignon committed:
[ADD] fs_product_multi_image: a drop-in replacement of storage_image_product

Commit:
[UPD] Update image_tag.pot

OCA-git-bot committed:
[BOT] post-merge updates

lmignon committed:
[IMP] image_tag: Removes dependency on sale module

Commit:
[UPD] Update image_tag.pot

OCA-git-bot committed:
[BOT] post-merge updates

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (13 of 13 strings)

Translation: storage-16.0/storage-16.0-image_tag
Translate-URL: https://translation.odoo-community.org/projects/storage-16-0/storage-16-0-image_tag/es/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (13 of 13 strings)

Translation: storage-16.0/storage-16.0-image_tag
Translate-URL: https://translation.odoo-community.org/projects/storage-16-0/storage-16-0-image_tag/it/

nguyenminhchien committed:
[IMP] image_tag: pre-commit auto fixes

nguyenminhchien committed:
[MIG] image_tag: Migration to 17.0

Commit:
[UPD] Update image_tag.pot

OCA-git-bot committed:
[BOT] post-merge updates

ChrisOForgeFlow committed:
[IMP] storage_import_image_advanced: : pre-commit stuff

ChrisOForgeFlow committed:
[16.0][MIG-REF] fs_import_image_advanced
- Refactor from storage_import_image_advanced
- Change to user fs_storage and fs_product_multi_image as dependencies
- Fix test to use properly test class dependency

ChrisOForgeFlow committed:
[ADD] fs_import_image_advanced_thumbnail

rousseldenis committed:
[IMP] image_tag: pre-commit autofixes

rousseldenis committed:
[MIG] image_tag: Migration to 18.0

lmignon committed:
[ADD] fs_field: Provides fields for filesystem contents

lmignon committed:
[IMP] fs_field: Add api to be used by the FsFolder widget

lmignon committed:
[IMP] fs_field: Add documentation

lmignon committed:
[IMP] fs_field: Add cleanup method in case of rollback

lmignon committed:
[IMP] fs_field: Add a way to sanitize the path used to create a folder

lmignon committed:
[FIX] fs_field: makes tests consistent from one environment to another

lmignon committed:
[IMP] fs_field: Add missing api methods

lmignon committed:
[FIX] fs_field: doc

sbidoul committed:
[DOC] fs_field: fix doc typos

astirpe committed:
[IMP] fs_attachment: test attachment with newline in filename

[IMP] fs_attachment: test attachment with newline in filename

xaviedoanhduy committed:
[IMP] fs_import_image_advanced: pre-commit auto fixes

xaviedoanhduy committed:
[IMP] fs_import_image_advanced_thumbnail: pre-commit auto fixes

xaviedoanhduy committed:
[MIG] fs_import_image_advanced: Migration to 18.0

xaviedoanhduy committed:
[MIG] fs_import_image_advanced_thumbnail: Migration to 18.0

xaviedoanhduy committed:
[DONT MERGE] test-requirements.txt

etobella committed:
[FIX] fs_content: Make preview work

nguyenminhchien commented on setup/fs_import_image_advanced/setup.py:
this file and its parent directory `setup` should not be added here.

xaviedoanhduy replied:
strange, if it come from this commit then it is not a migration commit

![image](https://github.com/user-attachments/assets/a48bedea-486c-4585-aea1-5d1e7d2414bd)

nguyenminhchien replied:
that folder should be deleted by pre-commit command. you could run that command again, and if it's still there, you could remove it yourself, then, commit the changes with `--amend ` to pre-commit commit.

xaviedoanhduy replied:
tried with `pre-commit run -a` but it didn't seem to work as expected so I removed it manually. 
Is it ok for you now?

nguyenminhchien commented on test-requirements.txt:
```diff
@@ -1,3 +1,8 @@
 requests_mock
 vcrpy-unittest
+odoo-addon-fs_image @ git+https://github.com/OCA/storage.git@refs/pull/446/head#subdirectory=fs_image
+odoo-addon-fs_image_thumbnail @ git+https://github.com/xaviedoanhduy/storage.git@refs/pull/1/head#subdirectory=fs_image_thumbnail
```
@xaviedoanhduy remember to replace this one (and `odoo-addon-fs_base_multi_image` below) by oca's PRs when creating PR to oca repository.

etobella committed:
[ADD] fs_field_demo

etobella committed:
[IMP] fs_field: Add JS

etobella committed:
[IMP] fs_field: Pass logic to a service in order to use it in other places

etobella committed:
[IMP] fs_field: Allow to initialize from the widget directly

etobella committed:
[FIX] fs_field_demo: Hide button

etobella committed:
[IMP] fs_field: Improve views. Add headers

etobella committed:
[FIX] fs_field: Fix upload of files

etobella committed:
[IMP] fs_field: Remove double components. improve buttons

etobella committed:
[IMP] fs_field: Allow to rename folders & files

etobella committed:
[IMP] fs_field: Improve management of icons

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (45 of 45 strings)

Translation: storage-18.0/storage-18.0-storage_thumbnail
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-storage_thumbnail/it/

etobella committed:
[IMP] fs_field: Improve table management

etobella committed:
[FIX] fs_field: Make button work

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (67 of 67 strings)

Translation: storage-18.0/storage-18.0-fs_storage
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-fs_storage/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (23 of 23 strings)

Translation: storage-18.0/storage-18.0-storage_backend
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-storage_backend/it/

etobella committed:
[IMP] fs_field: Allow sort of files

etobella committed:
[IMP] fs_field: Allow to hide columns

etobella committed:
[FIX] fs_field: Use proper names

etobella committed:
[IMP] fs_field: Improve view

lmignon committed:
[ADD] fs_storage_ms_drive: Microsoft Drive File System

This addon in addition to the microsoft_drive_account addon addon
extends the functionality of the fs_storage addon module to allow the
system to use Microsoft Drives (ie OneDrive, Sharepoint) as a storage backend.

etobella committed:
[FIX] fs_field: use onWillRender

etobella committed:
[IMP] fs_field: Add names. Fix directory icon

lmignon committed:
[IMP] fs_field: Improves sorting

* Sort char and text using locale and in a case and accent incensitive way
* When soring on name, keep directory first into the list
* Allows to define a comparison method into the field definition used if specified into the sort method

lmignon committed:
[FIX] fs_field: Pass the right parameter to the adapter

etobella committed:
[IMP] fs_field: Use effect. initialize some states

etobella committed:
Merge pull request #1 from acsone/18.0-js-lmi

Some improvements

lmignon committed:
[ADD] fs_folder_ms_drive: FsFolder extension for microsoft drive file system

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (23 of 23 strings)

Translation: storage-18.0/storage-18.0-storage_backend
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-storage_backend/it/

Commit:
[UPD] Update storage_backend_sftp.pot

Commit:
[UPD] Update storage_backend_s3.pot

Commit:
[UPD] Update storage_backend_ftp.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: storage-18.0/storage-18.0-storage_backend_ftp
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-storage_backend_ftp/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: storage-18.0/storage-18.0-storage_backend_sftp
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-storage_backend_sftp/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: storage-18.0/storage-18.0-storage_backend_s3
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-storage_backend_s3/

lmignon committed:
Move fs_folder and fs_folder_demo from incubator

lmignon committed:
[IMP] fs_storage_*: renamed from fs_field_*

lmignon committed:
[IMP] fs_folder_ms_drive: Redirect to login if not connected

OCA-git-bot committed:
[UPD] addons table in README.md

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (14 of 14 strings)

Translation: storage-18.0/storage-18.0-storage_backend_sftp
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-storage_backend_sftp/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (20 of 20 strings)

Translation: storage-18.0/storage-18.0-storage_backend_s3
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-storage_backend_s3/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (21 of 21 strings)

Translation: storage-18.0/storage-18.0-storage_backend_ftp
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-storage_backend_ftp/it/

lmignon committed:
[IMP] fs_folder: Proper tests isolation

When creating a storage used to store fs folder content, disbale possible existing one created by others addons

lmignon committed:
[FIX] fs_folder: call to method initialize on FsFolderValue instance must return a new instance

lmignon committed:
[IMP] fs_folder: Get access to the fs_storage instance from FsFolder value

lmignon committed:
[IMP] fs_folder: Allows to notify the widged when the content is modified

This action is usefull when an action from within odoo update the content and the widget need to be refreshed

lmignon committed:
[IMP] fs_folder: Avoid error in tests

Uses the TransactionCase class in place of BaseCommon as base class for tests to avoid error when tests are run on an database where the account module is installed but not loaded into the registry. The account addon adds a required column on the res.partner model with a default value but if the module is not loaded into the registry when a res.partner record is created, the creation fails with an integrity error

lmignon committed:
[IMP] fs_folder_ms_drive: Get scope from model 'mircrosoft.service'

lmignon committed:
[IMP] fs_storage_ms_drive: Update ms token as super user

pierre-halleux committed:
[IMP] microsoft_drive_account: avoid setting token validity if there is no token

vincent-hatakeyama committed:
[IMP] fs_storage: Allow setting check_connection_method in configuration file

vvrossem committed:
[IMP] storage_media: add archived ribbon and search filters

- storage_media_view_tree: add decoration-muted to visually indicate archived records.
- storage_media_view_form: display "Archived" ribbon when the record is inactive.
- storage_media_view_search: add a search view with filters for "All" and "Archived" records.

OCA-git-bot committed:
Merge PR #488 into 18.0

Signed-off-by simahawk

Commit:
[UPD] Update storage_media.pot

Commit:
[UPD] Update storage_image_product.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: storage-18.0/storage-18.0-storage_image_product
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-storage_image_product/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: storage-18.0/storage-18.0-storage_media
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-storage_media/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (43 of 43 strings)

Translation: storage-18.0/storage-18.0-storage_media
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-storage_media/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (44 of 44 strings)

Translation: storage-18.0/storage-18.0-storage_image_product
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-storage_image_product/it/

sbidoul committed:
[FIX] fs_attachment: domain to store attachment in db was always [(1, '=', 1)]

This is due to domain simplification algorithm introduced recently in Odoo.

OCA-git-bot committed:
Merge PR #489 into 18.0

Signed-off-by lmignon

OCA-git-bot committed:
[BOT] post-merge updates

sbidoul committed:
[IMP] fs_attachment: make force_storage_to_db_for_special_fields more flexible

By allowing to select the storage on which
to apply the move, it gives more flexibility.
For instance, when migrating an existing filestore to fs_attachment,
it is common to have all attachements in fsprod://, but wanting to apply
force_storage_to_db_for_special_fields
in a test environment where the default
storage is not fsprod.

sbidoul committed:
[IMP] fs_attachment: add some logging

simahawk committed:
storage_backend_s3: make regions static

These regions are not changing since ages. Pointless to keep fetching them at boot.

lmignon committed:
[IMP] fs_storage: Force mimetype for s3 only if provided

OCA-git-bot committed:
Merge PR #496 into 18.0

Signed-off-by lmignon

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #485 into 18.0

Signed-off-by etobella

Commit:
[UPD] Update fs_storage.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: storage-18.0/storage-18.0-fs_storage
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-fs_storage/

OCA-git-bot committed:
Merge PR #493 into 18.0

Signed-off-by simahawk

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (69 of 69 strings)

Translation: storage-18.0/storage-18.0-fs_storage
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-fs_storage/it/

lmignon committed:
[FIX] fs_attachment: Simplify code

The implementation of the method for calculating the URL for accessing attachments from an external file system was unnecessarily complex, mainly due to a bug in the fsspec library, which has since been fixed from fspesc>=2025.3.0

lmignon committed:
[FIX] fs_attachment: Makes the calculation of the x-accel-redirect path overridable.

Depending on the type of file system used, it may be necessary to adapt the way the path is calculated for file access via x-accel-redirect. A good example is the need to generate pre-signed URLs to allow unauthenticated access to files stored in an S3  protected by authentication.

lmignon committed:
[FIX] fs_attachment_s3: Specific features for fs_attachment into S3

This module extends the functionality of [fs_attachment](https://github.com/OCA/storage/tree/16.0/fs_attachment)
to better support Amazon S3 storage. It includes features such as signed URLs for X-Accel-Redirect, and enforcing
the mimetype of files stored in S3.

OCA-git-bot committed:
Merge PR #492 into 18.0

Signed-off-by lmignon

OCA-git-bot committed:
[BOT] post-merge updates

lmignon committed:
[IMP] fs_attachment: Improve X-Sendfile configuration handling

* Decouples the use of the X-Sendfile mechanism for externally stored content from the global X-Sendfile setting in Odoo.

* Previously, X-Sendfile had to be enabled both globally in Odoo and on each filesystem storage.

* With this change, Odoo no longer needs to have X-Sendfile enabled globally to allow its use at the external filesystem level.

lmignon committed:
[IMP] fs_attachment, fs_attachment_s3: x-send-file: Introduce generic management mechanism

Previously, enabling x-send-file required defining a dedicated redirection rule in the reverse proxy for each storage backend configured to use it.

This change introduces a generic approach that allows configuring a single reverse proxy rule applicable to all storages. The path used in X-Accel-Redirect now follows a standardized format:

The reverse proxy can parse this path to reconstruct the target URL and serve the file accordingly. This simplifies configuration and improves maintainability.

lmignon committed:
[IMP] fs_attachment: Restore original format for x-accel-redirect

The new variable “x_send_file_base_url” has also been removed to avoid unnecessarily complicating the code without any real commercial necessity.

lmignon committed:
[IMP] fs_attachement_s3: Simplfy x-accel-redirect path generation

lmignon committed:
[IMP] fs_attachment: X-Sendfile on S3 storage doesn't require a base url

lmignon committed:
[IMP] fs_attachment_s3: doc

lmignon committed:
[IMP] fs_attachment, fs_attachement_s3: Code simplification

lmignon committed:
[IMP] fs_attachment_s3: Improves doc

Co-authored-by: Stéphane Bidoul <stephane.bidoul@acsone.eu>

lmignon committed:
[IMP] fs_attachment, fs_attachment_s3: Naming consistency

Always uses x_sendfile into the code to refer to the x-senfile/x-accel-redirect mechanism

lmignon committed:
[FIX] fs_attachement_s3: Fix method naming

lmignon committed:
[FIX] fs_folder: fix sanitize method

The hyphen is a valid character in the name of a directory or file.

lmignon committed:
[IMP] fs_folder_ms_drive: Add column description

lmignon committed:
[IMP] fs_folder: Improve breadcrumb display

lmignon committed:
[FIX] microsoft_drive_account: Neutralize script must reset connexion state

Before this change, Microsoft connection data became inconsistent after running the database neutralization script. The problem was caused by the fact that only some of the data was reset. Among the data that was not reset was, for example, the connection status field.

lmignon committed:
[IMP] fs_attachment_s3: Better python dependency

lmignon committed:
[FIX] fs_folder_ms_dive: Fix preview url

If the storage is configured with a root path, if must prefix the path given when you call the preview method on the backend

lmignon committed:
[MIG] image_tag: Fix list view

Co-authored-by: youring <4132418+youring@users.noreply.github.com>

OCA-git-bot committed:
Merge PR #448 into 18.0

Signed-off-by lmignon

Commit:
[UPD] Update image_tag.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #480 into 18.0

Signed-off-by lmignon

Commit:
[UPD] Update fs_folder.pot

Commit:
[UPD] Update fs_folder_demo.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #481 into 18.0

Signed-off-by lmignon

Commit:
[UPD] Update fs_storage_ms_drive.pot

Commit:
[UPD] Update fs_folder_ms_drive.pot

Commit:
[UPD] Update microsoft_drive_account.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: storage-18.0/storage-18.0-fs_storage_ms_drive
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-fs_storage_ms_drive/it/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (3 of 3 strings)

Translation: storage-18.0/storage-18.0-fs_folder_demo
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-fs_folder_demo/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: storage-18.0/storage-18.0-fs_folder_ms_drive
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-fs_folder_ms_drive/it/

lmignon committed:
[MIG] fs_attachment, fs_attachment_s3: Forward port from 16.0

lmignon committed:
[MIG] fs_attachment, fs_attachment_s3: Forward port from 17.0

OCA-git-bot committed:
Merge PR #512 into 18.0

Signed-off-by lmignon

Commit:
[UPD] Update fs_attachment_s3.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (41 of 41 strings)

Translation: storage-18.0/storage-18.0-fs_folder
Translate-URL: https://translation.odoo-community.org/projects/storage-18-0/storage-18-0-fs_folder/it/

mymage committed:
Added translation using Weblate (Italian)
